pkg/metrics/exporter.go

```diff
@@ -0,0 +1,87 @@
+package metrics
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"sync"
+
+	"go.opencensus.io/stats/view"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	curMetricsExporter view.Exporter
+	metricsMux         sync.RWMutex
+)
+
+var (
+	metricsBackend = flag.String("metrics-backend", "Prometheus", "Backend used for metrics")
+	prometheusPort = flag.Int("prometheus-port", 8888, "Prometheus port for metrics backend")
+)
+
+const prometheusExporter = "prometheus"
+
+type Runner struct {
+	mgr manager.Manager
+}
+
+func AddToManager(m manager.Manager) error {
+	mr, err := New(m)
+	if err != nil {
+		return err
+	}
+	return m.Add(mr)
+}
+
+func New(mgr manager.Manager) (*Runner, error) {
+	mm := &Runner{
+		mgr: mgr,
+	}
+	return mm, nil
+}
+
+// Start implements the Runnable interface
```

Comment:
We can enforce this guarantee via the type system by adding:


```
var _ manager.Runnable = &Runner{}
```

Reply:
thanks, added